### PR TITLE
[release/8.0.1xx-rc1] Update with internal flow

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

Update RC1 with internal flow

- Updates internal scripts from Arcade.

### CODE REVIEW

File: eng/pipelines/common/pack.yml [hunk 1]
```diff
@@ -33,17 +33,63 @@ parameters:
...
 
 steps:
-  - template: provision.yml
-    parameters:
-      checkoutDirectory: ${{ parameters.checkoutDirectory }}
-      poolName: ${{ parameters.poolName }}
-      provisionatorChannel: ${{ parameters.provisionatorChannel }}
-      gitHubToken: ${{ parameters.gitHubToken }}
+
+  - ${{ if ne(variables['Build.DefinitionName'], 'dotnet-maui') }}:
```
Comment (rmarinho) [inline eng/pipelines/common/pack.yml:37]: this is to allow builds on dnceng the doesn't like provisionator

File: eng/pipelines/common/pack.yml [hunk 2]
```diff
@@ -33,17 +33,63 @@ parameters:
...
+        checkoutDirectory: ${{ parameters.checkoutDirectory }}
+        poolName: ${{ parameters.poolName }}
+        provisionatorChannel: ${{ parameters.provisionatorChannel }}
+        gitHubToken: ${{ parameters.gitHubToken }}
 
   - ${{ each step in parameters.prepareSteps }}:
     - ${{ each pair in step }}:
         ${{ pair.key }}: ${{ pair.value }}
 
+  - task: PowerShell@2
```
Comment (rmarinho) [inline eng/pipelines/common/pack.yml:49]: Setup the internal feeds using arcade stuff

File: eng/pipelines/common/pack.yml [hunk 3]
```diff
@@ -33,17 +33,63 @@ parameters:
...
+    condition: eq(variables['PrivateBuild'], 'true')
+    displayName: Setup Private Feeds Credentials
+    inputs:
+      filePath: $(System.DefaultWorkingDirectory)/eng/common/SetupNugetSources.ps1
+      arguments: -ConfigFile $(System.DefaultWorkingDirectory)/NuGet.config -Password $Env:Token
+    env:
+      Token: $(dn-bot-dnceng-artifact-feeds-rw) 
+      
+  - pwsh: |
```
Comment (rmarinho) [inline eng/pipelines/common/pack.yml:58]: this on provisinator yaml, but if we skip that we need to do it here to make java compiler happy building the aar

File: eng/pipelines/maui-release-internal.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
+trigger:
```
Comment (rmarinho) [inline eng/pipelines/maui-release-internal.yml:1]: this is to be used on dnceng

**Review by mattleibow — DISMISSED** [review]

**Review by samhouts — APPROVED** [review]

**Pull request merged**